nginx-controller/controller/controller.go

```diff
@@ -1067,3 +1097,131 @@ func (lbc *LoadBalancerController) ValidateSecret(secret *api_v1.Secret) error {
 
 	return fmt.Errorf("Secret is not a TLS or JWK secret")
 }
+
+// getAllIngress returns a list of all ingress resources
+func (lbc *LoadBalancerController) getAllIngress() []nginx.IngressEx {
+	ings, err := lbc.ingLister.List()
+	if err != nil {
+		glog.Errorf("Error Obtaining Ingress Resources: %v", err)
+	}
+
+	ingExs := []nginx.IngressEx{}
+	for i, _ := range ings.Items {
+		ingEx, err := lbc.createIngress(&ings.Items[i])
+		if err != nil {
+			glog.Errorf("Error creating ingress resource %v: %v", ingEx.Ingress.Name, err)
+			continue
+		}
+		ingExs = append(ingExs, *ingEx)
+	}
+
+	return ingExs
+}
+
+func (lbc *LoadBalancerController) validateMergableType(ingEx *nginx.IngressEx) ([]nginx.IngressEx, error) {
+	if len(ingEx.Ingress.Spec.Rules) != 1 {
+		err := fmt.Errorf("Ingress Resources with the 'nginx.org/mergible-ingress-type' annotation must contain only one host")
+		return nil, err
+	}
+
+	if ingEx.MergeableType == "master" {
+		if ingEx.Ingress.Spec.Rules[0].HTTP != nil {
+			err := fmt.Errorf("Ingress Resources with the 'nginx.org/mergible-ingress-type' annotation set to 'master' cannot contain Paths")
+			return nil, err
+		}
+
+		for key, val := range ingEx.Ingress.Annotations {
+			if strings.Contains(key, "nginx.org/mergible-ingress-type") {
+				continue
+			}
+			if strings.Contains(key, "nginx.org/") {
+				if strings.Contains(val, "serviceName=") {
+					err := fmt.Errorf("Ingress Resources with the 'nginx.org/mergible-ingress-type' annotation set to 'master' cannot contain annotations with the 'serviceName' parameter. Annotation: %s:%s", key, val)
+					return nil, err
+				}
+			}
+		}
+
+	} else if ingEx.MergeableType == "minion" {
+		if len(ingEx.TLSSecrets) > 0 || ingEx.JWTKey != nil {
+			err := fmt.Errorf("Ingress Resources with the 'nginx.org/mergible-ingress-type' annotation set to 'minion' cannot contain TLSSecrets or JWTKeys")
+			return nil, err
+		}
+
+		if ingEx.Ingress.Spec.Rules[0].HTTP == nil {
+			err := fmt.Errorf("Ingress Resources with the 'nginx.org/mergible-ingress-type' annotation set to 'minion' must contain a Path")
+			return nil, err
+		}
+
+		for key, val := range ingEx.Ingress.Annotations {
+			if strings.Contains(key, "nginx.org/mergible-ingress-type") {
+				continue
+			}
+			if strings.Contains(key, "nginx.org/") {
```

Comment:
Let's in the first version of this feature ignore all annotations in a minion for simplicity and then we can add validation in the next version.  For now a minion will get all annotations from the master. What do you think?